### PR TITLE
Add ChecksSelectionHints in the ExecutionResults

### PR DESCRIPTION
**Pull request by EMaksy**
# Description

A selected cluster should show a check selection hints, in the "Show Results" view,  when no tests were executed and selected.

## How was this tested?

Added test for ChecksSelectionHints.
Additional test will be added later.

### CODE REVIEW

File: assets/js/components/ChecksResults/ChecksSelectionHints.jsx [hunk 1]
```diff
@@ -36,7 +39,11 @@ function ChecksSelectionHints({ clusterId, selectedChecks }) {
             <Button
               className="bg-waterhole-blue mx-auto px-2 py-2 w-1/4 xs:w-full"
               onClick={() => {
-                navigate(`/clusters/${clusterId}/settings`);
+                navigate(
+                  usingNewChecksEngine
+                    ? `/clusters_new/${clusterId}/settings`
+                    : `/clusters/${clusterId}/settings`
+                );
```
Comment (arbulu89) [inline assets/js/components/ChecksResults/ChecksSelectionHints.jsx:46]: Here in this file, below, we need to add the `usingNewChecksEngine` to the `TriggerChecksExecutionRequest` component

File: assets/js/components/ChecksResults/ChecksSelectionHints.test.jsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
+import React from 'react';
```
Comment (arbulu89) [inline assets/js/components/ChecksResults/ChecksSelectionHints.test.jsx:1]: I wouldn't create this test. As it is now, this logic will be removed soon and it simply tests the difference in the url.
If we want to check this component (which would be out of scope for this PR), we should test something else

File: assets/js/components/ExecutionResults/ExecutionResults.jsx [hunk 3]
```diff
@@ -121,9 +122,10 @@ function ExecutionResults({
       <ResultsContainer
         catalogError={false}
         clusterID={clusterID}
-        hasAlreadyChecksResults
+        hasAlreadyChecksResults={!!(executionData || executionLoading)}
         selectedChecks={checkResults}
         onCatalogRefresh={onCatalogRefresh}
+        usingNewChecksEngine="true"
```
Comment (arbulu89) [inline assets/js/components/ExecutionResults/ExecutionResults.jsx:128]: Change this by simply `usingNewChecksEngine`. That `"true"` is not even a real boolean hehe

File: assets/js/components/ExecutionResults/ExecutionResults.jsx [hunk 4]
```diff
@@ -121,9 +122,10 @@ function ExecutionResults({
       <ResultsContainer
         catalogError={false}
         clusterID={clusterID}
-        hasAlreadyChecksResults
+        hasAlreadyChecksResults={!!(executionData || executionLoading)}
```
Comment (arbulu89) [inline assets/js/components/ExecutionResults/ExecutionResults.jsx:125]: The next line `selectedChecks`, we would need to change it.
This way, you are using current check results, but that field is not for that, it is to get the currently selected checks.
To make it proper, we would need to get the selected checks in the `ExecutionResultsPage`, like `cluster?.selected_checks`, pass them to this view, and use them in that param

**Review by arbulu89 — CHANGES_REQUESTED** [review]
Hey @EMaksy 
Added some comments.
Besides that, I think we need other kind of testing. We would need to add a new test to the `ExecutionResults` view, where we need to test the result when we give `executionData=null` and `executionLoading=false`. In this scenario, we need to test that we render the hint view, checking some basic text like `It looks like there is no recent execution for current cluster` or similar

File: assets/js/components/ExecutionResults/ExecutionResults.jsx (hunk 3 above)
Comment (arbulu89) [reply assets/js/components/ExecutionResults/ExecutionResults.jsx:128]: Yes, that's why you only need to put `usingNewChecksEngine` without anything as value:
```
<ResultsContainer
        catalogError={false}
        clusterID={clusterID}
        hasAlreadyChecksResults={!!(executionData || executionLoading)}
        usingNewChecksEngine
>
```

Comment (EMaksy) [reply assets/js/components/ExecutionResults/ExecutionResults.jsx:128]: Ah it's always true then... 
Thanks!

File: assets/js/components/ChecksResults/ChecksSelectionHints.test.jsx (hunk 2 above)
Comment (EMaksy) [reply assets/js/components/ChecksResults/ChecksSelectionHints.test.jsx:1]: Oki i will remove this test

File: assets/js/components/ExecutionResults/ExecutionResults.jsx (hunk 4 above)
Comment (EMaksy) [reply assets/js/components/ExecutionResults/ExecutionResults.jsx:126]: Done that,  ExecutionResultPage passes me the selected checks to the ExecutionResults

File: assets/js/components/ChecksResults/ChecksSelectionHints.jsx (hunk 1 above)
Comment (EMaksy) [reply assets/js/components/ChecksResults/ChecksSelectionHints.jsx:46]: You mean to pass usingNewChecksEngine as prop to TriggerChecksExecutionRequest? 
Or something else ?

Comment (arbulu89) [reply assets/js/components/ChecksResults/ChecksSelectionHints.jsx:46]: Yes, exactly that

File: assets/js/components/ExecutionResults/ExecutionResultsPage.jsx [hunk 5]
```diff
@@ -38,17 +37,13 @@ function ExecutionResultsPage() {
   if (!cluster) {
     return <div>Loading...</div>;
   }
+  const clusterSelectedChecks = cluster?.selected_checks;
```
Comment (arbulu89) [inline assets/js/components/ExecutionResults/ExecutionResultsPage.jsx:40]: I don't think we need a new constant here. Simply pass `cluster?.selected_checks` to `clusterSelectedChecks` as we do with other props

Comment (EMaksy) [reply assets/js/components/ExecutionResults/ExecutionResultsPage.jsx:40]: Oki yeah!

Comment (EMaksy) [reply assets/js/components/ExecutionResults/ExecutionResultsPage.jsx:40]: Yeah sure :D

**Review by arbulu89 — APPROVED** [review]
Thanks @EMaksy 
Fantastic!

File: assets/js/components/ExecutionResults/ExecutionResults.test.jsx [hunk 6]
```diff
@@ -1,6 +1,7 @@
 import React from 'react';
 import { screen } from '@testing-library/react';
-
+import '@testing-library/jest-dom/extend-expect';
```
Comment (dottorblaster) [inline assets/js/components/ExecutionResults/ExecutionResults.test.jsx:3]: Could you move this to the bottom of these imports?

**Review by dottorblaster — CHANGES_REQUESTED** [review]
Just a tiny thing then LGTM

File: assets/js/components/ExecutionResults/ExecutionResults.test.jsx (hunk 6 above)
Comment (EMaksy) [reply assets/js/components/ExecutionResults/ExecutionResults.test.jsx:3]: Sure :zap:

**Review by dottorblaster — APPROVED** [review]
LGTM
